src/stores/media/index.ts

```diff
@@ -391,5 +392,20 @@ export const useMediaStore = defineStore('media', {
         throw new Error(errorMessage)
       }
     },
+
+    setMediaProperties(
+      type: SupportedMediaType,
+      id: string,
+      properties: Partial<DetailFromMediaType<typeof type>>
+    ) {
+      const item = this.getItemById(type, id)
+      if (item) {
+        Object.assign(item, properties)
```

Comment:
I didn't realize that you could update the store objects like this!

Reply:
I didn't either! I was pleasantly surprised that it worked. But it makes sense, I suppose, as Object.assign mutates the existing object which is a Vue reactive (I think?) or at least part of a deep-reactive tree...

Reply:
The fact that it works is even more bizarre because [the Vue docs](https://v2.vuejs.org/v2/guide/reactivity.html#For-Objects) state that properties added using `Object.assign` will not be reactive. Maybe that's changed in the composition API 🤷?

Reply:
Strange. I'm doing something wrong in this code sandbox because nothing is working as expected, but when I try `Object.assign(obj, { [count]: count ])` in the `doMutation` function it doesn't cause the computed to update. When I do `obj[count] = count` however, it also doesn't cause the computed to update...

https://codesandbox.io/s/relaxed-black-nqhos7?file=/src/components/HelloWorld.vue

Any ideas? I could definitely see Object.assign bypassing the Proxy that Vue uses to track reactive objects...

Reply:
Okay, I realized that what I was doing in that previous example was slightly different. I've updated it with something more representative (watching a specific property of the obj) and it works.

Maybe the composition API changed this after all.